vibrato/src/dictionary.rs

```diff
@@ -165,4 +167,66 @@ impl Dictionary {
             need_check: false,
         })
     }
+
+    /// Resets the user dictionary from a reader.
+    ///
+    /// # Arguments
+    ///
+    ///  - `user_lexicon_rdr`: A reader of a lexicon file `*.csv` in the MeCab format.
+    ///                        If `None`, clear the current user dictionary.
+    ///
+    /// # Errors
+    ///
+    /// [`VibratoError`] is returned when an input format is invalid.
+    pub fn user_lexicon_from_reader<R>(mut self, user_lexicon_rdr: Option<R>) -> Result<Self>
+    where
+        R: Read,
+    {
+        if let Some(user_lexicon_rdr) = user_lexicon_rdr {
+            let mut user_lexicon = Lexicon::from_reader(user_lexicon_rdr, LexType::User)?;
+            if let Some(mapper) = self.data.mapper.as_ref() {
+                user_lexicon.do_mapping(mapper);
+            }
+            if !user_lexicon.verify(self.connector()) {
+                return Err(VibratoError::invalid_argument(
+                    "user_lexicon_rdr",
+                    "user_lexicon_rdr includes invalid connection ids.",
```

Comment:
This error message is redundant.
```suggestion
                    "includes invalid connection ids.",
```